src/classes/AffiliationMappingsHealthCheckVMapper.cls

```diff
@@ -0,0 +1,305 @@
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Health Check
+* @group-content ../../ApexDocContent/HealthCheck.htm
+* @description Generate a HealthCheckGroup view model representing the results of
+* an Affiliation Mappings health check.
+*/
+public virtual with sharing class AffiliationMappingsHealthCheckVMapper {
+    /**********************************************************************************
+    * @description Instance for Singleton Pattern
+    **********************************************************************************/
+    @TestVisible
+    private static AffiliationMappingsHealthCheckVMapper instance;
+
+    /**********************************************************************************
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************************/
+    protected AffiliationMappingsHealthCheckVMapper() {}
+
+    /**********************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of AffiliationMappingsHealthCheckVMapper.
+    **********************************************************************************/
+    public static AffiliationMappingsHealthCheckVMapper getInstance() {
+        if (instance == null) {
+            instance = new AffiliationMappingsHealthCheckVMapper();
+        }
+
+        return instance;
+    }
+
+    /**********************************************************************************
+    * @description Retrieve the results for a for an AffiliationMappings health check.
+    * @return A HealthCheckGroupVModel representing a HealtCheckDescription for Affiliations.
+    **********************************************************************************/ 
+    public virtual HealthCheckGroupVModel getHealthCheckGroup() {
+        HealthCheckGroupVModel parentHealthCheckGroupVModel = new HealthCheckGroupVModel(
+            Label.stgHealthCheckAffiliationMappings,
+            Label.stgHealthCheckStatusPassed,
+            0,
+            0,
+            new List<String>(),
+            new List<HealthCheckItemVModel>()
+        );
+
+        Map<Id, RecordTypeModel> accRTModelsById = this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType);
+        List<AffiliationMappingsModel> affiliationMappingsModelsList = this.locateAffiliationMappingsService().getAffiliationMappingsModels();
+
+        for (AffiliationMappingsModel affiliationMappingsModel : affiliationMappingsModelsList) {
+            HealthCheckGroupVModel childAffiliationMappingsHealthCheckResults = this.getAffiliationMappingsHealthCheckResults(
```

Comment:
I am not entirely comfortable with this line. Do you need to add more parameters when adding new health checks? Wil that break our unit tests?

Reply:
This should retrieve the aggregate group for an individual affiliation mapping

Reply:
@xinnanlisf yes, I added the necessary parameters in the subsequent PRs for the additional health checks.  The unit tests will need to be updated accordingly.

Reply:
yeah that seems like a pain when adding new health checks and a new parameter is required. We will have to update tons of lines in unit tests.

Reply:
Let's add the extra parameters in this PR and provide empty maps, etc to them for now. That way we don't have to worry about it as much downstream and we're in good shape for our unit tests.